packages/components/src/tooltip/test/index.js

```diff
@@ -23,171 +28,136 @@ describe( 'Tooltip', () => {
 	describe( '#render()', () => {
 		it( 'should not render the tooltip if multiple children are passed', () => {
 			render(
-				<Tooltip text="Help text">
-					<button>Button 1</button>
-					<button>Button 2</button>
+				<Tooltip { ...props }>
+					<Button>Button 1</Button>
+					<Button>Button 2</Button>
 				</Tooltip>
 			);
 
-			const button = screen.getByText( 'Button 1' );
-			act( () => button.focus() );
-			expect( screen.queryByText( 'Help text' ) ).not.toBeInTheDocument();
+			expect(
+				screen.queryByText( 'tooltip text' )
+			).not.toBeInTheDocument();
 		} );
 
-		it( 'should render children', () => {
+		it( 'should not render the tooltip if there is no focus', () => {
 			render(
-				<Tooltip text="Help text">
-					<button>Hover Me!</button>
+				<Tooltip { ...props }>
+					<Button>Hover Me!</Button>
 				</Tooltip>
 			);
 
 			expect(
 				screen.getByRole( 'button', { name: 'Hover Me!' } )
 			).toBeVisible();
-			expect( screen.queryByText( 'Help text' ) ).not.toBeInTheDocument();
+			expect(
+				screen.queryByText( 'tooltip text' )
+			).not.toBeInTheDocument();
 		} );
 
-		it( 'should render children with additional tooltip when focused', async () => {
-			const mockOnFocus = jest.fn();
+		it( 'should render the tooltip when focusing anchor via tab', async () => {
+			const user = userEvent.setup();
 
 			render(
-				<Tooltip text="Help text">
-					<button onFocus={ mockOnFocus }>Hover Me!</button>
+				<Tooltip { ...props }>
+					<Button>Hover Me!</Button>
 				</Tooltip>
 			);
 
-			const button = screen.getByRole( 'button', { name: 'Hover Me!' } );
-			expect( button ).toBeVisible();
+			expect( document.body ).toHaveFocus();
 
-			// Before focus, the tooltip is not shown.
-			expect( screen.queryByText( 'Help text' ) ).not.toBeInTheDocument();
+			await user.tab();
 
-			act( () => button.focus() );
+			expect(
+				screen.getByRole( 'button', { name: /Hover me!/i } )
+			).toHaveFocus();
 
-			// Tooltip is shown after focusing the anchor.
-			const tooltip = screen.getByText( 'Help text' );
-			expect( tooltip ).toBeVisible();
-			expect( mockOnFocus ).toHaveBeenCalledWith(
-				expect.objectContaining( {
-					type: 'focus',
-				} )
```

Comment:
This was also testing that our custom `onFocus` was being called. I don't see why it wouldn't be called 🤔 But at the same time, I wanted to be sure that we're intentionally removing this assertion.

Reply:
I did not realize we had a custom `onFocus`. So I thought asserting that the button was in focus was all that was necessary. The event handlers in this version of tooltip are a bit confusing to me, to be honest. 

So I'm curious, what would be a scenario where asserting that the button`.toHaveFocus()` wouldn't find that the mock focus would?

Reply:
Thanks for taking a second look! ❤️